src/workerd/api/streams/queue.c++

```diff
@@ -389,6 +389,11 @@ void ByteQueue::ByobRequest::respond(jsg::Lock& js, size_t amount) {
   // those extra bytes and push them into the consumers queue so they can be picked
   // up by the next read.
   req.pullInto.filled += amount;
+  if (req.pullInto.atLeast > amount) {
```

Comment:
this is duplicated throughout and is rather error prone. How about introduce a new method to pullInto? Something like `req.pullInto.fill(amount)`? (not sure about the name)

Reply:
Actually, looking it over again, most of these were defensive and not needed since the read is being resolved immediately. I'm updating the PR. There should only be one place the calculation is necessary